src/app/screens/confirmBrc20Transaction/index.tsx

```diff
@@ -0,0 +1,138 @@
+import AccountHeaderComponent from '@components/accountHeader';
+import BigNumber from 'bignumber.js';
+import BottomBar from '@components/tabBar';
+import styled from 'styled-components';
+import useDebounce from '@hooks/useDebounce';
+import useWalletSelector from '@hooks/useWalletSelector';
+import { BRC20ErrorCode } from '@secretkeylabs/xverse-core/transactions/brc20';
+import { brc20TransferEstimateFees, CoreError } from '@secretkeylabs/xverse-core';
+import {
+  getFeeValuesForBrc20OneStepTransfer,
+  ConfirmBrc20TransferState,
+  ExecuteBrc20TransferState,
+} from '@utils/brc20';
+import { isInOptions, isLedgerAccount } from '@utils/helper';
+import { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useResetUserFlow } from '@hooks/useResetUserFlow';
+import { useTranslation } from 'react-i18next';
+import ConfirmBrc20TransactionComponent from './confirmBrc20TransactionComponent';
+import { OnChangeFeeRate } from './editFees';
+
+const ScrollContainer = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  overflow-y: auto;
+  &::-webkit-scrollbar {
+    display: none;
+  }
+  height: 600px;
+  width: 360px;
+  margin: auto;
+`;
+
+export function ConfirmBrc20Transaction() {
+  const { t } = useTranslation('translation', { keyPrefix: 'CONFIRM_BRC_20' });
+  const { selectedAccount } = useWalletSelector();
+  const navigate = useNavigate();
+  const {
+    recipientAddress,
+    estimateFeesParams,
+    estimatedFees: initEstimatedFees,
+    token,
+  }: ConfirmBrc20TransferState = useLocation().state;
+  const { subscribeToResetUserFlow } = useResetUserFlow();
+  useEffect(() => subscribeToResetUserFlow('/confirm-brc20-tx'), [subscribeToResetUserFlow]);
```

Comment:
Note that `useResetUserFlow` doesn't actually memoise `subscribeToResetUserFlow`, so each render results in the useEffect firing and the subscription being closed and re-opened. 

Also, in testing, I noticed that when this happened, the channel is closed on the useEffect rerendering, and when we run the subscribe again, we are subscribing to a closed channel because nothing opens it again. That results in the reset message never actually firing.

Reply:
It might actually be better to refactor that component to have a single arg of `path` and internally use a single useEffect where it creates a channel, subscribes as per subscribeToResetUserFlow with the input path, and returns the close channel function, that way it's all contained in a single useEffect with no race conditions.

Reply:
right! it makes more sense without the subscribeToResetUserFlow as an effect dep, so removed that first. I also added a TODO to refactor the useResetUserFlow, but I remember I wanted to keep it flexible in case the use case required more control

Reply:
I'm not sure the extra flexibility would ever be used and it makes the component a lot easier to misuse. 

e.g. if some one does this:
`useEffect(() => { subscribeToResetUserFlow('/confirm-brc20-tx') }, []);`

instead of this:
`useEffect(() => subscribeToResetUserFlow('/confirm-brc20-tx'), []);`

Then we're opening up to a potential memory leak ( a very slow one, but it's there none the less).

But if we were to move everything into the component, we would just have to do:
`subscribeToResetUserFlow('/confirm-brc20-tx')`

and the component would clean up after itself when it's unloaded.

Reply:
ok I can get behind that. the broadcast and close function would be moved to its own hook in that case

Reply:
Yeah, something like this:
```
export const useResetUserFlow = (path: UserFlowConfigKey) => {
  const navigate = useNavigate();

  useEffect(() => {
    const broadcastChannel = new BroadcastChannel(resetUserFlowChannel);
    broadcastChannel.onmessage = (message) => {
      if (message.data !== 'reset') {
        return;
      }

      // TODO: infer UserFlowConfigKey from location?
      if (!userFlowConfig[path]) {
        return;
      }
      navigate(userFlowConfig[path]?.resetTo);
    };

    return () => {
      broadcastChannel.close();
    };
  }, []);
};

export const broadcastResetUserFlow = () => {
  const broadcastChannel = new BroadcastChannel(resetUserFlowChannel);
  broadcastChannel.postMessage('reset');
  broadcastChannel.close();
};
```

Just note that this will result in the popup broadcasting the message would also receive it in this case, which I don't think is major.

Reply:
indeed it simplified the usage a lot. thanks!